app/services/user_invite_service.rb

```diff
@@ -0,0 +1,15 @@
+module UserInviteService
+  # @param name [String]
+  # @param email [String]
+  # @param roles [Array<Symbol>]
+  # @param resource [ApplicationRecord]
+  # @return [User]
+  def self.invite(email:, name: nil, roles: [], resource: nil)
```

Comment:
```suggestion
  def self.invite!(email:, name: nil, roles: [], resource: nil)
```
I think we ought to name this with a `!` in front so that we know it can raise an error.

Reply:
The `!` methods are semantic and mean different things at different times. In Rails-world, `!` (e.g. `save!` vs `save`) means "will raise an error" vs. "will return nil". But in Ruby-world, e.g. `merge!` vs `merge`, the `!` means "will mutate the object rather than return a new object". More particularly, there is never a `!` method without a corresponding method that doesn't have the `!`.

Any method can raise an error, even if it isn't marked with `!`. Any code calling a method should be able to handle exceptions coming from it.